javascript/javascript1/week2/smart-ease2.js

```diff
@@ -0,0 +1,135 @@
+//Flight booking fullname function
+
+function getFullname (firstname, surname, useFormalName) {
+    if (useFormalName == true) {
+        return "Lord " + firstname + " " + surname;
+    }
+    else {
+        return firstname + " " + surname;
+    }
+}
+
+//two variables assigned to getFullname function.
+let fullname1 = getFullname("Math", "Anton", true);
+let fullname2= getFullname("Mia", "David");
+
+//Log out the two fullname variables.
+console.log(fullname1);
+console.log(fullname2);
+
+
+//What do we do if useFormalName is not given as an argument?
+//But what if the person is a woman? Describe how you would fix the getFullname so it also works for women
+//below u can see what i did in those cases.
+
+/*
+function getFullname(firstName, surName, useFormalName, gender) {
+    if (typeof firstName !== 'string' || firstName === '') {
+       throw new Error('First name must be a non-empty string');
+    }
+     if (typeof surName !== 'string' || surName === '') {
+        throw new Error('last name must be a non-empty string');
+     }
+     
+    if (useFormalName) {
+
+        //question here, how can i specify (male and female) instead of string, so If something else is written except(male and female), error message shows
+         if (typeof gender !== 'string' || gender === '') {
+            throw new Error('gender must be a non-empty string');
+         }
+    
+        else if(gender==="male"){
+        return `Lord ${firstName} ${surName}`;
+        }
+        else if(gender==="female"){
+        return `Lady ${firstName} ${surName}`;
+        }
+    }
+    else {
+        return `${firstName} ${surName}`;
+    }
+}
+console.log(getFullname("Jad", "Sam", "male", "male"));//Log out 
+*/
+
+//Event application
+
+function getEventWeekday(Daysback) {
+    const day = new Date();
+    const weekDays=["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+    for (i = 0; i < weekDays.length; i++) {
+        if(Daysback>7){
+            day.setDate(Daysback + weekDays[2]);
+        return (weekDays[(day.setDate() + i) % 7]);
+
+        }else if(Daysback<7){
+        day.setDate(day.getDate() + Daysback);
+        return (weekDays[(day.getDay() + 4 + i) % 7]);
+        }
+    }
+}
```

Comment:
Check again the logic. For example, today is Thursday, `getEventWeekday(1);` should return `Friday`, but it returns `Tuesday`
Hint: `weekDays[(date.getDay() + days) % 7]`

Reply:
i have changed my code, and i will push it soon, not sure if i understood what they ask me to do exactly.